components/brave_perf_predictor/browser/p3a_bandwidth_savings_tracker.cc

```diff
@@ -0,0 +1,40 @@
+/* Copyright 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_perf_predictor/browser/p3a_bandwidth_savings_tracker.h"
+
+#include <numeric>
+#include <utility>
+
+#include "base/logging.h"
+#include "base/metrics/histogram_macros.h"
+#include "brave/components/brave_perf_predictor/browser/p3a_bandwidth_savings_permanent_state.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "components/prefs/scoped_user_pref_update.h"
+
+namespace brave_perf_predictor {
+
+P3ABandwidthSavingsTracker::P3ABandwidthSavingsTracker(PrefService* user_prefs)
+    : user_prefs_(user_prefs) {}
+
+void P3ABandwidthSavingsTracker::RecordSaving(uint64_t saving) {
+  if (saving > 0) {
+    // TODO @AndriusA: optimise if needed, loading permanent state on every
+    // record could be costly
+    auto* permanent_state = new P3ABandwidthSavingsPermanentState(user_prefs_);
```

Comment:
why new/delete here? Wouldn't it be better to keep unique_ptr in the class for this?

Reply:
even if you did want to create a new one each time for some reason, it would be better to either create a stack var or a unique_ptr in this scope that will auto delete